src/python/gudhi/representations/vector_methods.py

```diff
@@ -478,7 +506,11 @@ def transform(self, X):
             diagram, num_pts_in_diag = X[i], X[i].shape[0]
             pers = 0.5 * (diagram[:,1]-diagram[:,0])
             min_pers = np.minimum(pers,np.transpose(pers))
-            distances = DistanceMetric.get_metric("chebyshev").pairwise(diagram)
+            # Works fine with sklearn 1.0, but an ValueError exception is thrown on past versions
+            try:
+                distances = DistanceMetric.get_metric("chebyshev").pairwise(diagram)
+            except ValueError:
+                distances = np.empty(shape = [0, 0])
```

Comment:
That's only supposed to happen if diagram is empty, right? Then, for documentation purposes, could we add `assert len(diagram) == 0` inside this `except`?

Reply:
Smart move ! Like that we avoid to catch all `ValueError`.
Done on 37d7743